File: lib/alarm_event.dart
class AlarmEvent {
  String title;
  String description;
  int hour;
  int minutes;
  List<int> days;

  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {};
    map['title'] = title;
    map['description'] = description;
    map['hour'] = hour;
    map['minutes'] = minutes;
    map['days'] = days;
    return map;
  }
}

File: android/src/main/java/com/sia/alarm_plugin/AlarmEvent.java
package com.sia.alarm_plugin;

import java.util.ArrayList;
import java.util.Map;

/**
 * Created by  on 2021/5/1.
 */
public class AlarmEvent {
  String title;
  String description;
  int hour;
  int minutes;
  ArrayList<Integer> days;

  public static AlarmEvent parse(Map<String, Object> json) {
    AlarmEvent event = new AlarmEvent();
    event.title = (String) json.get("title");
    event.description = (String) json.get("description");
    event.hour = (int) json.get("hour");
    event.minutes = (int) json.get("minutes");
    event.days = (ArrayList<Integer>) json.get("days");
    return event;
  }
}
